src/jdk.compiler/share/classes/com/sun/tools/javac/util/JCDiagnostic.java

```diff
@@ -503,6 +509,9 @@ public abstract static class DiagnosticInfo {
         /** The diagnostic kind (i.e. error). */
         DiagnosticType type;
 
+        /** A set of diagnostic flags to be automatically added to newly created JCDiagnostics (if not null). */
```

Comment:
As commented elsewhere -- should we use `null` or something like an empty list/set?